### PR TITLE
Feature/Line chart: metrices and dimension UI and integration

### PR DESCRIPTION
**Pull request by rinku-kumar-psl**
### Description
[Describe what this change achieves]
Multi matrices and dimension UI Integration.
### Issues Resolved
[List any issues this PR will resolve]
https://github.com/opensearch-project/dashboards-observability/issues/125

### Check List
- [ ] Multi matrices and dimension UI Integration for line chart
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by spattnaik — CHANGES_REQUESTED** [review]

File: dashboards-observability/public/components/event_analytics/utils/utils.tsx [hunk 1]
```diff
@@ -285,3 +285,13 @@ export const findAutoInterval = (start: string = '', end: string = '') => {
 
   return [minInterval, [{ text: 'Auto', value: 'auto_' + minInterval }, ...TIME_INTERVAL_OPTIONS]];
 };
+
+// to convert hex color code to rgba format
```
[the remark this answers is not in the document]
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/event_analytics/utils/utils.tsx:289]: HI @spattnaik , Actually this rgb() method we already kept on the common utils file, and as per our use case we are customizing it every time for different charts. Updated it after rebasing with main.

File: dashboards-observability/public/components/visualizations/charts/lines/line.tsx [hunk 2]
```diff
@@ -20,15 +24,20 @@ export const Line = ({ visualizations, layout, config }: any) => {
...
   const lastIndex = fields.length - 1;
-  const mode =
-    dataConfig?.chartOptions && dataConfig.chartOptions.mode && dataConfig.chartOptions.mode[0]
-      ? dataConfig.chartOptions.mode[0].modeId
-      : 'line';
+
+  const mode = dataConfig?.chartStyles?.style || DefaultMode;
+  const lineShape = dataConfig?.chartStyles?.interpolation || Interpolation;
+  const lineWidth = dataConfig?.chartStyles?.lineWidth || LineWidth;
+  const showLegend = dataConfig?.legend?.showLegend && dataConfig.legend.showLegend !== ShowLegend ? false : true;
```
[the remark this answers is not in the document]
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/visualizations/charts/lines/line.tsx:37]: Thanks @spattnaik

File: dashboards-observability/public/components/visualizations/charts/lines/line.tsx [hunk 3]
```diff
@@ -37,21 +46,69 @@ export const Line = ({ visualizations, layout, config }: any) => {
...
+      };
+      const fillProperty = {
+        fill: 'tozeroy',
+        fillcolor: fillColor,
+      };
+      const multiYaxis = { yaxis: `y${index + 1}` };
+      multiMetrics = {
+        ...multiMetrics,
+        [`yaxis${index + 1}`]: {
+          // title: `yaxis${index + 1} title`,
```
[the remark this answers is not in the document]
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/visualizations/charts/lines/line.tsx:70]: sure!

**Comment by rinku-kumar-psl** [comment]
raising external PR

**Pull request closed without merging**